app/src/main/kotlin/com/wire/android/ui/home/conversations/edit/EditMessageMenuItems.kt

```diff
@@ -125,70 +124,117 @@ fun EditMessageMenuItems(
-                            )
-                        },
-                        title = stringResource(R.string.label_copy),
-                        onItemClick = onCopyItemClick
-                    )
+    if (message.expirationStatus is ExpirationStatus.Expirable) {
+        return EphemeralMessageEditMenuItems(
+            message = message,
+            onDetailsClick = onDetailsItemClick,
+            onDownloadAsset = onDownloadAssetClick,
+            onOpenAsset = onOpenAssetClick,
+            onDeleteMessage = onDeleteItemClick
+        )
+    } else {
+        return buildList {
+            if (isAvailable) {
+                add { ReactionOption(onReactionItemClick) }
+                add { MessageDetailsMenuOption(onDetailsItemClick) }
+                if (isCopyable) {
+                    add {
+                        MenuBottomSheetItem(
+                            icon = {
+                                MenuItemIcon(
+                                    id = R.drawable.ic_copy,
+                                    contentDescription = stringResource(R.string.content_description_copy_the_message),
+                                )
+                            },
+                            title = stringResource(R.string.label_copy),
+                            onItemClick = onCopyItemClick
+                        )
+                    }
                 }
-            }
-            if (!isEphemeral) add { ReplyMessageOption(onReplyItemClick) }
-            if (isAssetMessage) add { DownloadAssetExternallyOption(onDownloadAssetClick) }
-            if (isGenericAsset) add { OpenAssetExternallyOption(onOpenAssetClick) }
-            if (isEditable && !isEphemeral) {
-                add {
-                    MenuBottomSheetItem(
-                        icon = {
-                            MenuItemIcon(
-                                id = R.drawable.ic_edit,
-                                contentDescription = stringResource(R.string.content_description_edit_the_message)
+                    ReplyMessageOption(onReplyItemClick)
+                    if (isAssetMessage) add { DownloadAssetExternallyOption(onDownloadAssetClick) }
+                    if (isGenericAsset) add { OpenAssetExternallyOption(onOpenAssetClick) }
+                    if (isEditable) {
+                        add {
+                            MenuBottomSheetItem(
+                                icon = {
+                                    MenuItemIcon(
+                                        id = R.drawable.ic_edit,
+                                        contentDescription = stringResource(R.string.content_description_edit_the_message)
+                                    )
+                                },
+                                title = stringResource(R.string.label_edit),
+                                onItemClick = onEditItemClick
```

Comment:
Can we move this MenuBottomSheetItem to its own file, aka `EditMessageMenuOption`?